### PR TITLE
OADP-524, OADP-460 mtc operator type, noDefaultBackupLocation secret handling [1.0 cp] #701 #607

### PR DESCRIPTION
**Pull request by kaovilai**
1.0 cherrypick
#701  [OADP-524](https://issues.redhat.com//browse/OADP-524) mtc operator type
Enables alternative behavior when OADP is consumed by MTC

#607 [OADP-460](https://issues.redhat.com//browse/OADP-460) Don't get secret if NoDefaultBackupLocation

### CODE REVIEW

**Review by dymurray — APPROVED** [review]

File: controllers/validator_test.go [hunk 1]
```diff
@@ -47,6 +48,102 @@ func TestDPAReconciler_ValidateDataProtectionCR(t *testing.T) {
 					},
 				},
 			},
+			// objects: []client.Object{},
```
Comment (kaovilai) [inline controllers/validator_test.go:51]: removing  unused comment
```suggestion
```

File: controllers/validator_test.go [hunk 2]
```diff
@@ -50,6 +51,101 @@ func TestDPAReconciler_ValidateDataProtectionCR(t *testing.T) {
...
+			},
+			objects: []client.Object{
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cloud-credentials",
+						Namespace: "test-ns",
+					},
+				},
+			},
+			wantErr: false,
```
Comment (kaovilai) [inline controllers/validator_test.go:146]: cherry-pick error

**Review by kaovilai — COMMENTED** [review]
found a cherry-pick error

File: controllers/validator_test.go (hunk 2 above)
Comment (kaovilai) [reply controllers/validator_test.go:146]: looks like https://github.com/openshift/oadp-operator/pull/607 was not chery-picked to 1.0

**Review by dymurray — APPROVED** [review]

**Pull request merged**